docs/source/compiler.rst

```diff
@@ -325,25 +398,161 @@ rewiring, which is controlled by ``PRAGMA INITIAL_REWIRING``. The syntax is as f
+
+When no ``INITIAL_REWIRING`` pragma is provided the compiler will choose one of two options
+depending on the program:
+
++ ``NAIVE``: The qubits used in all instructions in the program satisfy the topological constraints of the device.
+
++ ``PARTIAL``: Otherwise.
+
+For example, if your program consists of two-qubit instructions where the qubits in each instruction are nearest neighbors on the device, the compiler will employ the native strategy:
+
+.. code:: python
+
+   from pyquil import Program, get_qc
+   from pyquil.gates import CZ
+
+   qc = get_qc("Aspen-8", as_qvm=True)
+   p = Program(CZ(3, 4))
+
+   print(qc.compile(p).program)
+
+   CZ 3 4
+
+In the above example, `CZ 3 4` touches qubits that are already nearest neighbours (and support a
+`CZ` instruction) and so the compiler employs the naive strategy (and thus does not rewire those
+qubits to use better ones).
+
+If however, the program uses qubits that `must` be rewired, then the compiler defaults to the
+partial strategy:
+
+.. code:: python
+
+   from pyquil import Program, get_qc
+   from pyquil.gates import CZ
+
+   qc = get_qc("Aspen-8", as_qvm=True)
+   p = Program(CZ(3, 4))
+
+   print(qc.compile(p).program)
+
+   RZ(-pi/2) 0
+   RX(pi/2) 0
+   RZ(-pi/2) 0
+   RZ(pi/2) 1
+   XY(pi) 1 0
+   RZ(pi/2) 1
+   RX(pi/2) 1
+   RZ(-pi/2) 1
+   XY(pi) 1 0
+   RZ(-pi/2) 0
+   RX(-pi/2) 0
+
+.. _naive_rewiring:
+
+NAIVE
+^^^^^
+
+In this mode, the compiler chooses the ``naive`` mapping between logical qubits and physical qubits,
+where logical qubit ``i`` is assigned to physical qubit ``i``. With this initial rewiring, the
+compiler will generally **not** move an instruction's qubits around even if it results in a poor
+execution fidelity. For example assume that ``Aspen-8`` has a low-fidelity ``CZ 0 1``, then
+compiling this program with naive rewiring will **not** move the ``CZ`` to a better qubit pair:
+
+.. code:: python
+
+   from pyquil import Program, get_qc
+   from pyquil.gates import CZ
+
+   qc = get_qc("Aspen-8", as_qvm=True)
+   p = Program('PRAGMA INITIAL_REWIRING "NAIVE"', CZ(0, 1))
+
+   print(qc.compile(p).program)
+
+   PRAGMA INITIAL_REWIRING "NAIVE"
+   CZ 0 1
+
+If, however, your program includes an instruction that does **not** use neighboring qubits the
+compiler will be required to insert swaps (virtual or real, see swaps_) that might affect the
+logical-physical qubit mapping. For example,
+
+.. code:: python
+
+   from pyquil import Program, get_qc
+   from pyquil.gates import CZ
+
+   qc = get_qc("Aspen-8", as_qvm=True)
+   p = Program('PRAGMA INITIAL_REWIRING "NAIVE"', CZ(0, 2))
+
+   print(qc.compile(p).program)
+
+   PRAGMA INITIAL_REWIRING "NAIVE"
+   CZ 6 5
+
+In the above program ``CZ 0 2`` is not a native instruction (meaning it cannot be directly executed
+on the target device) and so the compiler must insert a swap (virtual, in this case) into the
+program. When rewiring must occur in this mode it is **not** gauranteed that the resulting program
```

Comment:
sp: guaranteed

Reply:
Trying to spell that word gives me anxiety :(